src/store.c

```diff
@@ -0,0 +1,153 @@
+#include <sys/mman.h>
+#include <fcntl.h>
+#include <unistd.h>
+
+#include "utils.h"
+#include "error.h"
+#include "header.h"
+#include "store.h"
+#include "errno.h"
+
+#define HSK_HEADERS_BIN "/tmp/hnsd.bin";
+
+int
+hsk_store_init(hsk_store_t *store, const uv_loop_t *loop) {
+  store->loop = (uv_loop_t *)loop;
+  store->fd = -1;
+  store->init = false;
+  store->map = NULL;
+  store->location = HSK_HEADERS_BIN;
+  store->headers = malloc(sizeof(hsk_store_t));
+  if (!store->headers)
+    return HSK_ENOMEM;
+  store->timer = NULL;
+  return HSK_SUCCESS;
+}
+
+int hsk_store_uninit(hsk_store_t *store) {
+  store->map = NULL;
+  store->location = NULL;
+  store->headers = NULL;
+  if (munmap(store->map, store->size) == -1)
```

Comment:
I think we have to rethink the approach of using memory-mapped files. Originally I thought that since we handle re-orgs in memory, if we're mapping the headers from memory to the mmap file, we don't have to re-implement re-org when writing to the file. I'm not so sure if this works like I had planned. Also, mmap doesn't seem to work on windows and I'm not sure about BSD.

The alternative is to just use a normal file.